### PR TITLE
Test viewport tests

### PR DESCRIPTION
**Pull request by sfranky**

### CODE REVIEW

File: qtop.py [hunk 1]
```diff
@@ -1585,7 +1591,7 @@ def get_document(scheduling_system):
                         from xml.etree import ElementTree as etree
 
                 if options.TRANSPOSE:
-                    config['transpose_wn_matrices'] = 'False' if config['transpose_wn_matrices'] == 'True' else 'True'
+                    config['transpose_wn_matrices'] = not config['transpose_wn_matrices']
```
Comment (sfranky) [inline qtop.py]: @lookfwd remember the conversation we had about this? Now that I have eval'ed it early on, the A = Not A construct is possible. (and don't start yelling about evals, I know, I'll fix them in the future!!)

Comment (fgeorgatos) [inline qtop.py]: very good thing to be done with the multiple evals; 

generally, any type of code repetition should be subjected to scrutiny, no matter how small!

File: qtop.py [hunk 2]
```diff
@@ -1681,8 +1692,8 @@ def get_document(scheduling_system):
 
                 # justification for implementation:
                 # http://unix.stackexchange.com/questions/47407/cat-line-x-to-line-y-on-a-huge-file
-                line_offset = viewport.get_v_stop() - viewport.get_v_start()
-                cat_command = 'clear;tail -n+%s %s | head -n%s' % (viewport.get_v_start(), output_fp, line_offset)
+                # line_offset = viewport.v_stop - viewport.v_start
+                cat_command = 'clear;tail -n+%s %s | head -n%s' % (viewport.v_start, output_fp, viewport.get_v_term_size())
```
Comment (fgeorgatos) [inline qtop.py]: could we, perhaps, pre-cache this stream of characters and then dump it in one go?
(I see some flicker when on slow screen sessions, eg. remote ssh)

OR, piping the output via `cat` or `tee`, might just help it somewhat!

**Comment by fgeorgatos** [comment]
OK, overall !

**Pull request merged**